tests/e2e/system_test/test_reclaim_space_cronjob.py

```diff
@@ -0,0 +1,212 @@
+            # if for some reason the pvcs were not deleted in the test - they will be deleted here
+            for pvc_obj in self.pvc_objs_created:
+                pvc_obj.delete()
+
+            for temp_file in self.temp_files_list:
+                if os.path.exists(temp_file):
+                    run_cmd(f"rm {temp_file}")
+
+            run_oc_command(cmd=f"delete namespace {self.namespace}")
+
+        request.addfinalizer(finalizer)
+
+    def test_reclaim_space_cronjob(self):
+        """
+        Test case to check reclaim space cronjobs are created correctly for rbd pvcs in openshift-* namespace
+        """
+        timeout = 30
+        num_of_samples = 10
+        namespace = f"openshift-{uuid4().hex}"
+        self.namespace = namespace
+        res = run_oc_command(cmd=f"create namespace {self.namespace}")
+        assert ERRMSG not in res[0], (
+            f"Failed to create namespace with name {namespace}" f"got result: {res}"
+        )
+        logger.info(f"Namespace {namespace} created")
+
+        res = run_oc_command(cmd=f"get namespace {namespace} -o yaml")
+        namespace_dict = yaml.safe_load("\n".join(res))
+
+        schedule = namespace_dict["metadata"]["annotations"][
+            "reclaimspace.csiaddons.openshift.io/schedule"
+        ]
+        assert (
+            schedule == "@weekly"
+        ), f"Namespace {namespace} created with schedule {schedule}, expected @weekly"
+        logger.info(f"Existence of schedule {schedule} validated.")
+
+        for i in range(num_of_samples):
+            try:
+                pvc_obj = helpers.create_pvc(
+                    sc_name=constants.DEFAULT_STORAGECLASS_RBD,
+                    size="1Gi",
+                    namespace=namespace,
+                )
+                helpers.wait_for_resource_state(pvc_obj, constants.STATUS_BOUND)
+            except Exception as e:
+                logger.exception(f"The PVC was not created, exception [{str(e)}]")
+                raise PVCNotCreated("PVC did not reach BOUND state.")
+
+            logger.info(f"PVC by name {pvc_obj.name} created")
+            self.pvc_objs_created.append(pvc_obj)
+
+        # wait since it may take some time for cronjobs to be created
+        time.sleep(30)
+        res = run_oc_command(cmd="get reclaimspacecronjob", namespace=namespace)
+        logger.info(f"Reclaim space jobs after PVC creation {res}")
+        assert (
+            len(res) > 1
+        ), f"No reclaim space cron jobs exist in namespace {namespace}"
+        cronjob_names = []
+        for j in res[1:]:
+            cronjob_names.append(j.split()[0])
+
+        for pvc_obj in self.pvc_objs_created:
+            cronjob_for_pvc = [
+                name for name in cronjob_names if name.startswith(f"{pvc_obj.name}-")
+            ]
+            assert (
+                len(cronjob_for_pvc) == 1
+            ), f"No reclaim space cron job for pvc {pvc_obj.name} created"
```

Comment:
'cronjob_names' contains' the list of all cronjob names in this namespace. cronjob_for_pvc is the list of all cronjobs associated with the given PVC. This assert verifies that there exist exactly 1 such job so it seems to be correct. I agree that it may be not fully clear from the code so I've changed the assert message.